src/pool/rpc.h

```diff
@@ -604,6 +253,33 @@ pool_connect_in_set_data(crt_rpc_t *rpc, uint64_t pci_flags, uint64_t pci_query_
 	}
 }
 
+#define DAOS_ISEQ_POOL_DISCONNECT    /* input fields */ ((struct pool_op_in)(pdi_op)CRT_VAR)
+
+#define DAOS_ISEQ_POOL_DISCONNECT_V6 /* input fields */ ((struct pool_op_v6_in)(pdi_op)CRT_VAR)
+
+#define DAOS_OSEQ_POOL_DISCONNECT    /* output fields */ ((struct pool_op_out)(pdo_op)CRT_VAR)
+
+CRT_RPC_DECLARE(pool_disconnect, DAOS_ISEQ_POOL_DISCONNECT, DAOS_OSEQ_POOL_DISCONNECT)
+CRT_RPC_DECLARE(pool_disconnect_v6, DAOS_ISEQ_POOL_DISCONNECT_V6, DAOS_OSEQ_POOL_DISCONNECT)
+
+#define DAOS_ISEQ_POOL_QUERY /* input fields */                                                    \
+	((struct pool_op_in)(pqi_op)CRT_VAR)((crt_bulk_t)(pqi_map_bulk)CRT_VAR)(                   \
+	    (uint64_t)(pqi_query_bits)CRT_VAR)
+
+#define DAOS_ISEQ_POOL_QUERY_V6 /* input fields */                                                 \
+	((struct pool_op_v6_in)(pqi_op)CRT_VAR)((crt_bulk_t)(pqi_map_bulk)CRT_VAR)(                \
+	    (uint64_t)(pqi_query_bits)CRT_VAR)
+
+#define DAOS_OSEQ_POOL_QUERY /* output fields */                                                   \
+	((struct pool_op_out)(pqo_op)CRT_VAR)((daos_prop_t)(pqo_prop)CRT_PTR)(                     \
+	    (struct daos_pool_space)(pqo_space)CRT_RAW)(                                           \
+	    (struct daos_rebuild_status)(pqo_rebuild_st)CRT_RAW) /* only set on -DER_TRUNC */      \
```

Comment:
These macro looks hard to read, why not keep space between var?

Reply:
comes from clang-format. There may be a recipe available for clang-format to (temporarily) not flag these macros with the human-readable spacing that we need. I will fix them, but perhaps in a separate PR if it is too complicated to apply the technique along with the other changes I'm making in this patch.

Reply:
In a separate draft PR based on this branch, I've tried to manually fix up the macros affected by clang-format. The code there manually uses tabs to make the macros readable, and uses clang-format off / on comments to mark what sections of the header file should be ignored (or processed) by clang-format.

https://github.com/daos-stack/daos/pull/13440/